docs/user_defined_math/syntax.md

```diff
@@ -215,6 +215,21 @@ But if you're having trouble setting up your math, it is a useful function to ge
     Our internally defined parameters, listed in the `Parameters` section of our [pre-defined base math documentation][base-math] all have default values which propagate to the math.
     You only need to use `default_if_empty` for decision variables and global expressions, and for user-defined parameters.
 
+### where
```

Comment:
Not really, as it doesn't _need_ to subset. It has two functions: 
1. potentially applying a condition (may or may not be considered 'subsetting')
2. potentially extending the dimensions over which the math component is indexed

It can do one or the other, or both. 

I wonder if `group` might work better - it's effectively its purpose as described in #604 - or `extend_dims_and_apply_where` for maximum clarity & verbosity.

Reply:
Hmmm... I am not a fan of overly verbose names, and _extend_dims_and_apply_where_ would bloat our syntax too much. The fact that an _and_ is in it maybe hints at it doing too many things.

If this is essentially doing the same thing as the general syntax _where_, let us keep that name.

Reply:
Yeah, we could have a `broadcast` and a `where`, but since `where` automatically broadcasts in xarray, it's difficult to separate them. `where` everywhere else means `broadcast and apply where`, so I guess we stick with it.